### DISCUSSION TITLE
Google custom OAuth 2.0 client_id/secret not working - Example file in documentation wrong?

### DISCUSSION THREAD

**Discussion by markus22-22** [iterate-ch/cyberduck#15585] [Q&A] [answered]
Hi guys

The example .cyberduckprofile file linked in the public docs:
https://docs.duck.sh/protocols/profiles/google_client_id/

Still uses the OOB method. It doesn't work.

Does anyone have a working profile that I can use as a template?

Or what should I substitute it for?
I think this was discussed here but no reference to custom creds. for Google Drive mentioned.
https://github.com/iterate-ch/cyberduck/issues/13587

Many thanks.

**Comment by dkocher** [comment]
You will have to switch to use `${oauth.application.identifier}:oauth` for the _OAuth Redirect Url_.

```
        <key>OAuth Redirect Url</key>
        <string>${oauth.application.identifier}:oauth</string>
```

The `${oauth.application.identifier}` will be replaced on runtime with either `io.cyberduck` or `io.mountainduck` respectively.

**Comment by dkocher** [comment]
Examples will be updated with iterate-ch/docs#470.

**Comment by markus22-22** [comment]
Hey! Many thanks!!

It still doesn't work. I get "Error 400: redirect_uri_mismatch" from Google.
I am using CyberDuck Version 8.7.2 (40975) on macOS 13.5.1
I have deleted all previous test profiles from "Group Containers/G69SCX94XU.duck/Library/Application Support/duck".
Here's my .cyberduckprofile (with redacted id/secret).
[GDrive-custom.txt](https://github.com/iterate-ch/cyberduck/files/14066689/GDrive-custom.txt)

**Reply by dkocher** [reply]
Plesae confirm you have configured the redirect URI `x-cyberduck-action:oauth` in your app registration.

**Reply by markus22-22** [reply]
Hey! Thanks! Where should I configure that?

What I tried so far:
- GCP console: It is not on the consent screen area, and it is not on the credentials area when creating a set of creds. for a _"Desktop App"_. I tried to create a set of creds for a _"Web app"_, and indeed there's a place there to put a redirect URI. But it doesn't allow to use anything that is not a FQDN, it says _"Invalid Origin: must contain a domain."_

- Try to substitute the "\<key>OAuth Redirect Url\</key>" in the .cyberduckprofile from "_${oauth.application.identifier}_" to "_x-cyberduck-action:oauth_" in case that's what you meant. But it doesn't work either.

So not sure where to put that.

**Reply by markus22-22** [reply]
@dkocher any chance we can make this work? We'd really like to use Cyberduck in our organization.
Thanks in advance :)

**Comment by dkocher** [comment, marked as answer]
I have the following revised instructions to register a custom OAuth 2.0 Client ID in the [Google Cloud Console](https://console.cloud.google.com/apis/credentials).

 * Choose _Desktop app_ as the _Application type_ which will result in a Client ID with a suffix like `number-id.apps.googleusercontent.com`. 
 * Then set the `OAuth Redirect Url` in the connection profile like 

```
        <key>OAuth Redirect Url</key>
        <string>com.googleusercontent.apps.number-id:oauth</string>

```

**Reply by markus22-22** [reply]
It works!!! Thank you very much!
I think you should update the docs/example file with this.
If it wasn't for this thread it would have been impossible to guess if one is not a developer.
Thanks again.
